tools/socktap/tcp_link.cpp

```diff
@@ -31,6 +31,9 @@ void TcpLink::request(const access::DataRequest& request, std::unique_ptr<ChunkP
 
     // insert packet size before ethernet header
     uint16_t packet_size = packet->size() + EthernetHeader::length_bytes;
```

Comment:
I think the fix should be replacing `packet->size()` by `packet->size(OsiLayer::Network, OsiLayer::Application)`.

Reply:
Yes, that is a more elegant solution. I didn't know about this function.